### PR TITLE
Fix bugs after GDS creation refactoring

### PR DESCRIPTION
**Pull request by DarthMax**

### CODE REVIEW

File: graphdatascience/aura_graph_data_science.py [hunk 1]
```diff
@@ -26,43 +26,41 @@ def __init__(
...
-        self._driver_config = gds_query_runner.driver_config()
+        gds_query_runner = ArrowQueryRunner.create(
+            Neo4jQueryRunner.create(endpoint, auth, aura_ds=True),
+            auth,
+            True,
+            arrow_disable_server_verification,
+            arrow_tls_root_certs,
+        )
+
+        gds_query_runner.set_database("neo4j")
```
Comment (soerenreichardt) [inline graphdatascience/aura_graph_data_science.py:41]: This is not necessary as this is the default

File: graphdatascience/aura_graph_data_science.py [hunk 2]
```diff
@@ -26,43 +26,41 @@ def __init__(
         endpoint: str,
         auth: Tuple[str, str],
         aura_db_connection_info: AuraDbConnectionInfo,
-        database: Optional[str] = None,
```
Comment (soerenreichardt) [inline graphdatascience/aura_graph_data_science.py:29]: Why do you think that this is not necessary anymore? I think we need this

**Review by soerenreichardt — CHANGES_REQUESTED** [review]
I think agree with removing the database selection. Otherwise this looks fine

File: graphdatascience/aura_graph_data_science.py (hunk 1 above)
Comment (FlorentinD) [reply graphdatascience/aura_graph_data_science.py:41]: i dont see where this is the default. The default is `None`.

Comment (soerenreichardt) [reply graphdatascience/aura_graph_data_science.py:41]: https://github.com/neo4j/graph-data-science-client/blob/main/graphdatascience/query_runner/neo4j_query_runner.py#L83

Comment (soerenreichardt) [reply graphdatascience/aura_graph_data_science.py]: Still necessary as the default is `None` and not `neo4j` which is a bit counter-intuitive. I added a comment clarifying this

**Review by soerenreichardt — APPROVED** [review]

**Comment by soerenreichardt** [comment]
@DarthMax we thought a bit about how to set the database and so on and preferred to align the remote projection with what other procedures do by having to call `gds.set_database` instead of passing that as an explicit argument to the procedure

**Pull request merged**